app/src/main/java/com/simpragma/recipe/adapterclass/DataAdapter.java

```diff
@@ -0,0 +1,190 @@
+import java.util.List;
+
+/**
+ * Created by Madvesha  on 2/1/18.
+ */
+
+public class DataAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    Recipe recipeResult;
+    Context context;
+    private ArrayList<RecipeList> apiResultList;
+    private static final int ITEM = 0;
+    private static final int LOADING = 1;
+    private boolean isLoadingAdded = false;
+
+    public DataAdapter(Context context, Recipe recipeResult, ArrayList<RecipeList> arrylist) {
+        this.recipeResult = recipeResult;
+        this.context = context;
+        apiResultList = arrylist;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
+
+        RecyclerView.ViewHolder viewHolder = null;
+        LayoutInflater inflater = LayoutInflater.from(viewGroup.getContext());
+
+        switch (viewType) {
+            case ITEM:
+                viewHolder = getViewHolder(viewGroup, inflater);
+                break;
+            case LOADING:
+                View v2 = inflater.inflate(R.layout.item_progress, viewGroup, false);
+                viewHolder = new LoadingViewHolder(v2);
+                break;
+        }
+        return viewHolder;
+
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, final int i) {
+
+        switch (getItemViewType(i)) {
+
+            case ITEM:
+                final ViewHolder recipeVH = (ViewHolder) holder;
+                Log.d("DATAADAPTER", apiResultList.get(i).getTitle().trim());
+                recipeVH.titleTextView.setText(apiResultList.get(i).getTitle().trim());
+                Picasso.with(context).load(apiResultList.get(i).getThumbnail())
+                        .placeholder(R.mipmap.ic_launcher).resize(85, 85).into(recipeVH.recipeImageView);
+                break;
+            case LOADING:
+//               Do nothing
+                break;
+        }
+
+        holder.itemView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                // open another activity on item click
+                Intent browserIntent = new Intent(Intent.ACTION_VIEW);
+                browserIntent.setData(Uri.parse(apiResultList.get(i).getHref().toString().trim()));
+                context.startActivity(browserIntent);
+            }
+        });
+
+    }
+
+
+    @NonNull
+    private RecyclerView.ViewHolder getViewHolder(ViewGroup parent, LayoutInflater inflater) {
+        RecyclerView.ViewHolder viewHolder;
+        View v1 = inflater.inflate(R.layout.card_row, parent, false);
+        viewHolder = new ViewHolder(v1);
+        return viewHolder;
+    }
+
+
+    @Override
+    public int getItemCount() {
+        return apiResultList == null ? 0 : apiResultList.size();
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        return (position == apiResultList.size() - 1 && isLoadingAdded) ? LOADING : ITEM;
```

Comment:
What happens if apiResultList is null?

Reply:
When get response that time checking response size.
If response size is zero 
its display No recipe available for given ingredients
if response size is grater then zero
it comes to adapter class